extras/module-node/README.rst

```diff
@@ -0,0 +1,62 @@
+=============================
+Node.js-like module framework
+=============================
+
+This directory contains an example module resolution and loading framework and
+``require`` implementation based on Node.js.  The application need only
+provide the module resolution and loading logic:
+
+* Add ``duk_module_node.c`` to list of C sources to compile.
+
+* Ensure ``duk_module_node.h`` is in the include path.
+
+* Include the extra header in calling code and initialize the bindings::
+
+      #include "duktape.h"
+      #include "duk_module_node.h"
+
+      /* After initializing the Duktape heap or when creating a new
+       * thread with a new global environment:
+       */
+      duk_module_node_init(ctx, resolve_cb, load_cb);
+
+* The resolve callback looks like this::
+
+      duk_ret_t cb_resolve_module(duk_context *ctx) {
+          const char *module_id = duk_get_string(ctx, 0);
+          const char *parent_id = duk_get_string(ctx, 1);  /* calling module ID */
+          const char *resolved_id;
+
+          /*
+           *  Arrive at a canonical module ID in whatever way makes sense.
+           *  For Node.js emulation, it is recommended to use the fully
+           *  resolved filename of the module.
+           */
+
+          duk_push_string(ctx, resolved_id);
+          return 1;  /*nrets*/
+      }
+
+  Important note: When ``require()`` is called outside of any module, the
+  parent ID will be an empty string.
+
+* The load callback looks like this::
+
```

Comment:
Same here, mention that it's a Duktape/C function.

Maybe mention somewhere that both resolve and load can be Ecmascript functions too?